airbyte_cdk/sql/shared/catalog_providers.py

```diff
@@ -121,9 +121,12 @@ def get_primary_keys(
         stream_name: str,
     ) -> list[str]:
         """Return the primary keys for the given stream."""
-        pks = self.get_configured_stream_info(stream_name).primary_key
-        if not pks:
-            return []
+        configured_stream = self.get_configured_stream_info(stream_name)
```

Comment:
Let's expand the docstring slightly, to make the behavior explicit. Just add a blank line and then something like "We will use `primary_key` if it is set explicitly in the configured catalog, otherwise we will fall back to `source_defined_primary_key`, if set."